backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapper.java

```diff
@@ -0,0 +1,277 @@
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Material;
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Partner;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialPartnerRelationService;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialService;
+import org.eclipse.tractusx.puris.backend.stock.domain.model.MaterialItemStock;
+import org.eclipse.tractusx.puris.backend.stock.domain.model.ProductItemStock;
+import org.eclipse.tractusx.puris.backend.stock.domain.model.ReportedMaterialItemStock;
+import org.eclipse.tractusx.puris.backend.stock.domain.model.ReportedProductItemStock;
+import org.eclipse.tractusx.puris.backend.stock.logic.dto.itemstocksamm.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Service
+@Slf4j
+public class ItemStockSammMapper {
+
+    @Autowired
+    private MaterialService materialService;
+    @Autowired
+    private MaterialPartnerRelationService mprService;
+
+    public ItemStockSAMM toItemStockSAMM(MaterialItemStock materialItemStock) {
+        ItemStockSAMM samm = new ItemStockSAMM();
+        samm.setDirection(DirectionCharacteristic.INBOUND);
+        samm.setPositions(new ArrayList<>());
+        samm.setMaterialGlobalAssetId(materialItemStock.getMaterial().getMaterialNumberCx());
+        samm.setMaterialNumberCustomer(materialItemStock.getMaterial().getOwnMaterialNumber());
+        samm.setMaterialNumberSupplier(mprService.find(materialItemStock.getMaterial(),
+            materialItemStock.getPartner()).getPartnerMaterialNumber());
+        var posList = new ArrayList<Position>();
+        samm.setPositions(posList);
+        Position position = new Position();
+        if (materialItemStock.getCustomerOrderId() != null || materialItemStock.getCustomerOrderPositionId() != null
+            || materialItemStock.getSupplierOrderId() != null) {
+            OrderPositionReference opr = new OrderPositionReference(materialItemStock.getSupplierOrderId(),
+                materialItemStock.getCustomerOrderId(), materialItemStock.getCustomerOrderPositionId());
+            position.setOrderPositionReference(opr);
+        }
+        ItemQuantityEntity itemQuantityEntity = new ItemQuantityEntity(materialItemStock.getQuantity(),
+            materialItemStock.getMeasurementUnit());
+        AllocatedStock allocatedStock = new AllocatedStock(itemQuantityEntity, materialItemStock.getLocationBpns(),
+            materialItemStock.isBlocked(), materialItemStock.getLocationBpna());
+        var allocatedStocksList = new ArrayList<AllocatedStock>();
+        allocatedStocksList.add(allocatedStock);
+        position.setAllocatedStocks(allocatedStocksList);
+        return samm;
+    }
+
+    public ItemStockSAMM toItemStockSamm(ProductItemStock productItemStock) {
```

Comment:
did not evaluate this method in detail. But same applies to this method: 
- posList is not set to SAMM
- should get a list of productItemStocks